vector/src/main/java/fr/gouv/tchap/features/platform/TchapGetPlatformTask.kt

```diff
@@ -0,0 +1,112 @@
+import org.matrix.android.sdk.api.failure.Failure
+import org.matrix.android.sdk.api.failure.MatrixError
+import org.matrix.android.sdk.api.session.identity.ThreePid
+import timber.log.Timber
+import java.util.ArrayList
+import java.util.Random
+import javax.inject.Inject
+
+sealed class GetPlatformResult {
+    data class Success(val platform: Platform) : GetPlatformResult()
+    data class Failure(val throwable: Throwable) : GetPlatformResult()
+}
+
+data class Params(
+        val email: String
+)
+
+/**
+ * Get the Tchap platform configuration (HS/IS) for the provided email address.
+ */
+class TchapGetPlatformTask @Inject constructor(
+        private val context: Context,
+        private val activeSessionHolder: ActiveSessionHolder,
+        private val matrix: Matrix
+) : ViewModelTask<Params, GetPlatformResult> {
+
+    override suspend fun execute(params: Params): GetPlatformResult {
+        Timber.d("## TchapGetPlatformTask [${params.email}]")
+
+        // Prepare the list of the known IS urls in order to run over the list until to get an answer.
+        val idServerUrls = buildIdServerUrls().toMutableList()
+        if (idServerUrls.isEmpty()) {
+            return GetPlatformResult.Failure(Failure.ServerError(MatrixError(
+                    code = MatrixError.M_UNKNOWN,
+                    message = "No host"
+            ), 403))
+        }
+
+        var failure: Throwable? = null
+        idServerUrls.onEach { url ->
+            try {
+                val platform = matrix.threePidPlatformDiscoverService().getPlatform(url, ThreePid.Email(params.email))
+                Timber.d("## TchapGetPlatformTask succeeded (${platform.hs})")
+                return GetPlatformResult.Success(platform)
+            } catch (throwable: Throwable) {
+                Timber.e(throwable, "## TchapGetPlatformTask failed ")
+                failure = throwable
+            }
+        }
+
+        return GetPlatformResult.Failure(failure ?: Failure.Unknown(null))
+    }
+
+    private fun buildIdServerUrls(): List<String> {
+        val identityService = activeSessionHolder.getSafeActiveSession()?.identityService()
+        val idServerUrls: MutableList<String> = ArrayList()
+
+        // Consider first the current identity server if any.
+        val currentIdServerUrl = identityService?.getCurrentIdentityServerUrl()
+        if (currentIdServerUrl != null) {
+            idServerUrls.add(currentIdServerUrl)
+        }
+
+        // Add randomly the preferred known ISes
+        var currentHosts: MutableList<String> = mutableListOf(*context.resources.getStringArray(R.array.preferred_identity_server_names))
+        while (currentHosts.isNotEmpty()) {
+            val index = Random().nextInt(currentHosts.size)
+            val host: String = currentHosts.removeAt(index)
+            val idServerUrl = context.getString(R.string.server_url_prefix) + host
+            if (null == currentIdServerUrl || idServerUrl != currentIdServerUrl) {
```

Comment:
i prefer : "currentIdServerUrl == null"

Reply:
Me too, I moved this code from the PlatformViewModel and this is a convention that we find in many other places in the project

Reply:
You can ask @bmarty about this.

Reply:
👋 
Using `null == currentIdServerUrl` is an old Java convention, to avoid mistaking between:
`currentIdServerUrl == null` and
`currentIdServerUrl = null` which can lead to hard-to-see bugs.

I think in Kotlin there is no reason not to use `currentIdServerUrl == null`, since you cannot write 
```kotlin
if(currentIdServerUrl = null) {...}
```
or you will get the error `Assignments are not expressions, and only expressions are allowed in this context`

Reply:
Fixed